extensions/kafka/src/main/java/io/deephaven/kafka/ingest/KafkaIngester.java

```diff
@@ -404,4 +480,22 @@ public void shutdownPartition(final int partition) {
         }
         kafkaConsumer.wakeup();
     }
+
+    /**
+     * Add a callback for the consumer loop.
+     * 
+     * @param callback a callback that allows subscribers to inject logic into the consumer loop.
+     */
+    public void addConsumerLoopCallback(@NotNull final ConsumerLoopCallback callback) {
+        consumerLoopCallbacks.add(callback);
+    }
+
+    /**
+     * Remove a callback from the consumer loop.
+     * 
+     * @param callback a previously added callback.
+     */
+    public void removeConsumerLoopCallback(@NotNull final ConsumerLoopCallback callback) {
+        consumerLoopCallbacks.remove(callback);
+    }
```

Comment:
I'm curious, does DHE need dynamic add/remove callback support? Would `Set<ConsumerLoopCallback>` in constructor suffice?

Reply:
Good point!